### ISSUE TITLE
New PRs

### ISSUE THREAD

**Issue by jeroenvervaeke** [websockets-rs/rust-websocket#257]
Hi everyone!

When I opened this crate the readme states
> Note: Maintainership of this project is slugglish. You may want to use tungstenite or tokio-tungstenite instead.

Is it worth making pull requests for this project or do you suggest to move on to another project?

**Comment by vi** [comment]
I already started trying to make a special version of `rust-websocket` that tries to partially preserve the API of it, but uses `tungstenite` / `tokio-tungstenite` inside, to provide remaining users of `rust-websocket` nicer way to eventually migrate to it.

Do you see some value in `rust-websocket` codebase that is not covered by other alternative?

**Comment by vi** [comment]
What pull requests do you have plans to make? `tokio 1.0` support or something more drastic?

**Comment by jeroenvervaeke** [comment]
I wanted to attempt upgrading to `tokio 1.0` and migrate to futures `0.3.12` so the `futures::Stream` can be used. Basically migrating to more standard implementations so it's easier to use the library

**Comment by vi** [comment]
If decide to continue and it would the tests (maybe after fixing `.travis.yml`, if Travis still works) then I'll publish that release.

**Comment by jeroenvervaeke** [comment]
Ok, good to know, I'll give it a shot!

**Comment by vi** [comment]
I remember trying to do that myself (but for Tokio 0.2) and eventually abandoned the attempt. Maybe I was not trying hard enough.

**Comment by vi** [comment]
@jeroenvervaeke , Have you managed to make it work on Tokio 1?

**Comment by jeroenvervaeke** [comment]
I started working on it, but it was way too much work to "make it work on Tokio 1".
A total rewrite would probably be easier than refactoring it, a lot has changed.

For my own project I decided to use the sync `tungstenite` crate and send all messages over tokio channels.

**Comment by vi** [comment]
What advantages does this approach have over using `tokio-tungstenite`?

**Comment by jeroenvervaeke** [comment]
I started using `tokio-tungstenite` but some of the types I needed didn't implement `Send` or `Sync` (I don't recall which ones exactly).

My project was pretty simplistic, I had to listen to a websocket and process the messages, I did not need to "reply" to those messages. Switching to channels made it easier to unit test too.

This issue can be closed since I do not plan to continue working on the refactor. (Due to the above mentioned reasons)
